### PR TITLE
Add support for MacOS 13.3 SDK

### LINKED ISSUE #264: Support MACOSX_SDK_VERSION 13.3
### Comment:

When XCode [15.0 final](https://xcodereleases.com/) drops, it should be possible to support `MACOSX_SDK_VERSION: "13.3"`.

References:
- https://github.com/conda-forge/spidermonkey-feedstock/pull/11#issuecomment-1692277042

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

References:

- fixes #264

### CODE REVIEW

File: recipe/download_osx_sdk.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ export CONDA_BUILD_SYSROOT="${OSX_SDK_DIR}/MacOSX${MACOSX_SDK_VERSION}.sdk"
 if [[ ! -d ${CONDA_BUILD_SYSROOT} ]]; then
     echo "Downloading ${MACOSX_SDK_VERSION} sdk"
 
-    if [[ "${MACOSX_SDK_VERSION}" == "12.3" ]]; then
+    if [[ "${MACOSX_SDK_VERSION}" == "12.3" || "${MACOSX_SDK_VERSION}" == "13.3" ]]; then
```
Comment (isuruf) [inline recipe/download_osx_sdk.sh:26]: How about
```suggestion
    if [[ $(echo "${MACOSX_SDK_VERSION}" | cut -d "." -f 1) -gt 11 ]]; then
```
?

**Comment by bollwyvl** [comment]
I don't know much about this repo... anything else what needs doing? Tests?

**Comment by isuruf** [comment]
Can you add a `recipe/conda_build_config.yaml` with `MACOSX_SDK_VERSION: 13.3` and rerender? Then we should be able to see if it works.

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
Looks like it did the thing:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=786578&view=logs&j=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&t=63fa899e-c160-5185-75c4-c5fd30c90374&l=346

So presumably now remove that, and rerender?

**Comment by bollwyvl** [comment]
Should i be worried about the remaining changes? Just hard revert back to [fcafa88](https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/281/commits/fcafa8885d22f175e21f1b173ec04c3b61bbef38) ?

**Comment by beckermr** [comment]
They look fine to me!

**Comment by bollwyvl** [comment]
Thanks!

**Pull request merged**